src/transformers/models/textnet/convert_textnet_original_to_pytorch.py

```diff
@@ -0,0 +1,219 @@
+            backbone_config["stage2"]["stride"],
+            backbone_config["stage3"]["stride"],
+            backbone_config["stage4"]["stride"],
+        ],
+        hidden_sizes=[
+            config_dict["first_conv"]["out_channels"],
+            backbone_config["stage1"]["out_channels"][-1],
+            backbone_config["stage2"]["out_channels"][-1],
+            backbone_config["stage3"]["out_channels"][-1],
+            backbone_config["stage4"]["out_channels"][-1],
+        ],
+        out_features=["stage1", "stage2", "stage3", "stage4"],
+        out_indices=[1, 2, 3, 4],
+        image_size=size,
+    )
+
+    return textnet_config
+
+
+def convert_textnet_checkpoint(checkpoint_url, checkpoint_config_url, pytorch_dump_folder_path):
+    response = requests.get(checkpoint_config_url)
+    content = response.text
+    namespace = {}
+
+    exec(content, namespace)
+
+    model_config = namespace.get("model")
+    data_config = namespace.get("data")
+    size = 640
+    if "train" in data_config:
+        if "short_size" in data_config["train"]:
+            size = data_config["train"]["short_size"]
+
+    if "tiny" in model_config["backbone"]["config"]:
+        config = prepare_config(tiny_config_url, size)
+    elif "small" in model_config["backbone"]["config"]:
+        config = prepare_config(small_config_url, size)
+    else:
+        config = prepare_config(base_config_url, size)
+
+    model = TextNetBackbone(config)
+    textnet_image_processor = CLIPImageProcessor(
+        size={"shortest_edge": size},
+        resample=PILImageResampling.BILINEAR,
+        do_center_crop=False,
+        use_square_size=True,
+        image_mean=IMAGENET_DEFAULT_MEAN,
+        image_std=IMAGENET_DEFAULT_STD,
+    )
+    state_dict = torch.hub.load_state_dict_from_url(checkpoint_url, map_location="cpu", check_hash=True)["ema"]
+    state_dict_changed = OrderedDict()
+    for key in state_dict:
+        if "backbone" in key:
+            val = state_dict[key]
+            new_key = key
+            for search, replacement in rename_key_mappings.items():
+                if search in new_key:
+                    new_key = new_key.replace(search, replacement)
+
+            pattern = r"textnet\.stage(\d)"
+
+            def adjust_stage(match):
+                stage_number = int(match.group(1)) - 1
+                return f"textnet.encoder.stages.{stage_number}.stage"
+
+            # Using regex to find and replace the pattern in the string
+            new_key = re.sub(pattern, adjust_stage, new_key)
+            state_dict_changed[new_key] = val
+    model.load_state_dict(state_dict_changed)
+
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    image = Image.open(requests.get(url, stream=True).raw).convert("RGB")
+
+    # preprocess image
+    transformations = transforms.Compose(
+        [
+            transforms.Resize((size, size), interpolation=transforms.InterpolationMode.BILINEAR),
```

Comment:
could you link to the original implementation where they resize to a square? 

It seems like they resize only the shorter edge: https://github.com/czczup/FAST/blob/7ab105474ab226ab74a346b0648d2e2baab67d79/dataset/utils.py#L232